model-optimizer/extensions/front/RollWithEmptyAxesReplacer.py

```diff
@@ -0,0 +1,53 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+from extensions.front.AttributedRollToRoll import AttributedRollToRoll
+from extensions.ops.roll import Roll
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.common.replacement import FrontReplacementOp
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph, rename_nodes
+from mo.ops.const import Const
+from mo.ops.reshape import Reshape
+from mo.ops.shape import Shape
+
+
+class RollWithEmptyAxesReplacer(FrontReplacementOp):
+    """
+    This transformation replaces Roll with empty axes input with the following sequence of operations:
+    reshape to 1D tensor -> roll -> reshape to original shape.
+    """
+    enabled = True
+    op = "Roll"
+
+    def run_after(self):
+        return [AttributedRollToRoll]
+
+    def replace_sub_graph(self, graph: Graph, match: dict):
```

Comment:
Nit: actually I think overriding `find_and_replace_pattern` from `FrontReplacementPattern` is better as it was in your previous version. It will work even if you inherit from `FrontReplacementSubgraph`, as it was done below

https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/front/onnx/LoopNormalize.py#L26-L27

https://github.com/openvinotoolkit/openvino/blob/master/model-optimizer/extensions/front/kaldi/tdnn_component_replacer.py#L42-L43

But it's Nit, i do not insist to change your current solution.

Reply:
Done.